internal/gensupport/resumable.go

```diff
@@ -239,3 +225,33 @@ func (rx *ResumableUpload) Upload(ctx context.Context) (resp *http.Response, err
 		return prepareReturn(resp, err)
 	}
 }
+
+// shouldRetry indicates whether an error is retryable for the purposes of this
+// package, following guidance from
+// https://cloud.google.com/storage/docs/exponential-backoff .
+func shouldRetry(status int, err error) bool {
+	if 500 <= status && status <= 599 {
+		return true
+	}
+	if status == statusTooManyRequests {
+		return true
+	}
+	if err == io.ErrUnexpectedEOF {
+		return true
+	}
+	// Transient network errors should be retried.
+	if se, ok := err.(syscall.Errno); ok {
+		return se == syscall.ECONNRESET || se == syscall.ECONNREFUSED
```

Comment:
Need this behind a build flag.

Also, I think you can do:

```go
if err == syscall.ECONNRESET || err == syscall.ECONNREFUSED {
  // ...
}
```

no need to check the type first.

For the build flags, maybe something like

```go
// isRetryable is a platform-specific hook, specified in retryable_linux.go
var syscallRetryable func(error) bool

// ...

if syscallRetryable != nil && syscallRetryable(err) {
   return true
}
```
```go
// +build linux

func init() {
  syscallRetryable = func(err error) bool { return err == syscall.ECONNRESET || err == syscall.ECONNREFUSED }
}
```

Reply:
Thanks for the suggestion! Fixed.